File: baekjoon/shim/Python/단계별로_풀어보기/스택/n_10828.py
import sys
input=sys.stdin.readline
n=int(input())
s=[]
for _ in range(n):
    l=list(input().split())
    if l[0]=="push":
        s.append(int(l[1]))
    elif l[0]=="pop":
        print(-1 if len(s)==0 else s.pop())
    elif l[0]=="size":
        print(len(s))
    elif l[0]=="empty":
        print(1 if len(s)==0 else 0)
    else:
        print(-1 if len(s)==0 else s[-1])

File: baekjoon/shim/Java/단계별로_풀어보기/스택/n_10828.java
package Study_2022.baekjoon.shim.Java.단계별로_풀어보기.스택;

import java.io.*;
import java.util.*;

public class n_10828 {
    public static void main(String[] args)throws IOException{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw=new BufferedWriter(new OutputStreamWriter(System.out));
        int n=Integer.parseInt(br.readLine());
        Stack<Integer>stack=new Stack<>();
        for(int i=0; i<n; i++){
            String st[]=br.readLine().split(" ");
            switch(st[0]){
                case "push":
                    stack.push(Integer.parseInt(st[1]));
                    break;
                case "pop":
                    bw.write((stack.empty()?-1:stack.pop())+"\n");
                    break;
                case "size":
                    bw.write(stack.size()+"\n");
                    break;
                case "empty":
                    bw.write((stack.empty()?"1":"0")+"\n");
                    break;
                default:
                    bw.write((stack.empty()?"-1":stack.peek())+"\n");
                    break;
            }
        }
        bw.flush();bw.close();
    }
}
